src/entities/contract-agreement.ts

```diff
@@ -1,12 +1,19 @@
 import { Policy } from "./policy";
+import { JsonLdId, JsonLdValue } from "./jsonld"
 
-export interface ContractAgreement {
-  id: string;
-  assetId: string;
-  consumerAgentId: string;
+export class ContractAgreement extends JsonLdId {
+  'https://w3id.org/edc/v0.0.1/ns/assetId': JsonLdValue<string>[];
+  consumerAgentId?: string;
   contractEndDate?: number;
   constractSigningDate?: number;
   constractStartDate?: number;
-  policy: Policy;
-  providerAgentId: string;
+  policy?: Policy;
+  providerAgentId?: string;
+
+  get assetId(): string {
+    return this['https://w3id.org/edc/v0.0.1/ns/assetId']
+      .map(it => Object.assign(new JsonLdValue(), it))[0]
```

Comment:
If we want to be more safe, we can use `Array.prototype.at` instead.

```ts
  get assetId(): string {
    return this['https://w3id.org/edc/v0.0.1/ns/assetId']
      .map(it => Object.assign(new JsonLdValue(), it))
      .at(0)?
      .value
```

The return value would though change to `string | undefined`.

Perhaps, the thought I have in mind is: are we sure `assetId` is always present? (note: this thought is valid for all updated datatypes where we access data in this way)

Reply:
assetId is mandatory, in the future we'd need to distinguish between mandatory fields and optional, in case of mandatory ones, if they are missing, everything should explode